### PR TITLE
Konflux CI builds

### PR DESCRIPTION
**Pull request by svghadi**
This PR enables dex image build via Konflux CI. The PR and Push container image builds run in hermetic (disconnected) env. The Push builds create image for all supported arch i.e amd64, arm64, ppc64le & s390x whereas PR builds build image only on amd64 due to long delays in setting up build env for ppc64le & s390x.

### CODE REVIEW

File: Containerfile.plugin [hunk 1]
```diff
@@ -17,50 +17,58 @@
 # --- start build stage #1
 FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_8_golang_1.22 as builder
 
-COPY . . 
-WORKDIR /app
-ENV GOFLAGS=""
+COPY . /app
+WORKDIR /app/dex
 
-RUN GOOS=linux go generate -mod=mod /app/storage/ent/
```
Comment (svghadi) [inline Containerfile.plugin:24]: Removed the go generate STEP which causes dependency issues in hermetic env. The go generate step is not part of upstream Dockerfile. Additionally, it does't do any code change or generation so I believe it is safe to remove it from downstream aswell. The go generate in dex is similar to the bundle generation commands we run for operator. Here it generates db scehma which is committed to the git repo like we do for bundles.

Comment (svghadi) [reply Containerfile.plugin:24]: Ran some test. Looks good. Removal of `go generate` doesn't cause any functionality problems.

**Comment by svghadi** [comment]
The PR is ready for review. I have tested the konflux built image. Everything looks fine.

**Review by iam-veeramalla — APPROVED** [review]
LGTM, Thanks @svghadi

**Comment by svghadi** [comment]
Push job built multi-arch image successfully. 

```bash
$ podman manifest inspect quay.io/redhat-user-workloads/rh-openshift-gitops-tenant/dex:43b71c1fb4a8d42180f3cfc7ab1612ff0775fd0f
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 941,
            "digest": "sha256:8ba3f5ace899bb5e46e140e1949c75252474f7d0cb5f9c1b80892cecbb7480b3",
            "platform": {
                "architecture": "amd64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 941,
            "digest": "sha256:ed3e002d407046b88fb09aa988e1f47844c8ff61d525aa12144fe5b9744ece3a",
            "platform": {
                "architecture": "arm64",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 941,
            "digest": "sha256:95997cf4e8971a599fad7b4b39391dfca30893ccae79487f42fcb9ba45f1d7f3",
            "platform": {
                "architecture": "ppc64le",
                "os": "linux"
            }
        },
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "size": 941,
            "digest": "sha256:b5f6058b044b55f272df90c0fad1264abbca74a3c4e3d6bdcc98930de1483c4d",
            "platform": {
                "architecture": "s390x",
                "os": "linux"
            }
        }
    ]
}
```

Also the labels on image are as expected
```bash
$ podman inspect quay.io/redhat-user-workloads/rh-openshift-gitops-tenant/dex:43b71c1fb4a8d42180f3cfc7ab1612ff0775fd0f
[
........
               "Labels": {
                    "License": "Apache 2.0",
                    "architecture": "aarch64",
                    "build-date": "2024-11-28T06:49:06",
                    "com.redhat.component": "openshift-gitops-dex-container",
                    "com.redhat.delivery.appregistry": "false",
                    "com.redhat.license_terms": "https://www.redhat.com/en/about/red-hat-end-user-license-agreements#UBI",
                    "container-version": "43b71c1",
                    "description": "Red Hat OpenShift GitOps Dex",
                    "distribution-scope": "public",
                    "downstream-source-ref": "43b71c1fb4a8d42180f3cfc7ab1612ff0775fd0f",
                    "downstream-source-url": "https://github.com/rh-gitops-midstream/downstream-dex",
                    "downstream-vcs-type": "git",
                    "io.buildah.version": "1.38.0-dev",
                    "io.k8s.description": "The Universal Base Image Minimal is a stripped down image that uses microdnf as a package manager. This base image is freely redistributable, but Red Hat only supports Red Hat technologies through subscriptions for Red Hat products. This image is maintained by Red Hat and updated regularly.",
                    "io.k8s.display-name": "Red Hat OpenShift GitOps Dex",
                    "io.openshift.expose-services": "",
                    "io.openshift.tags": "openshift,gitops,dex,sso",
                    "maintainer": "William Tam <wtam@redhat.com>",
                    "name": "openshift-gitops-1/dex-rhel8",
                    "release": "1130",
                    "summary": "Red Hat OpenShift GitOps Dex",
                    "upstream-source-ref": "43956db7fd75c488a82c70cf231f44287300a75d",
                    "upstream-source-url": "https://github.com/dexidp/dex.git",
                    "upstream-vcs-type": "git",
                    "url": "https://access.redhat.com/containers/#/registry.access.redhat.com/ubi8-minimal/images/8.10-1130",
                    "vcs-ref": "43b71c1fb4a8d42180f3cfc7ab1612ff0775fd0f",
                    "vcs-type": "git",
                    "vendor": "Red Hat, Inc.",
                    "version": "8.10"
               }
          },
........
```

**Pull request merged**